spec/unit/lib/cloud_controller/diego/docker/lifecycle_protocol_spec.rb

```diff
@@ -118,6 +118,34 @@ module Docker
                 end
               end
             end
+
+            context 'when root user not allowed' do
+              before do
+                TestConfig.override(allow_docker_root_user: false)
+              end
+
+              context 'and the process does not set a user' do
```

Comment:
I _think_ we also want to block the case where:
1. `allowed_users` includes `root`
2. process/task user is set to `root`
3. `allow_docker_root_user` is set to `false`

This case that may never happen in the real world, but I think that's the correct behavior.